### ISSUE TITLE
Bug: math.flatten() clones object values too

### ISSUE THREAD

**Issue by manake** [josdejong/mathjs#2793]
```
let matrix = [[1, 2], [3, { key: 4 }]];
let flat = math.flatten(matrix);
```
It's fine that the above function returns a new array like `[1, 2, 3, { key: 4 }]` but it also unexpectedly clones `{ key: 4 }` instead of keeping it as a reference.

As the name of the function says: it should "flatten the matrix" and not "flatten the matrix AND change something about it".

IMO, in such case you should have `flatten` and `flattenDeep` if you really want to clone the objects inside matrices for some reason.

I'm pretty sure that current `flatten` will cause data loss in some cases. What if the value was `{ key1: someSingleton, key2: something.bind(this), key3: new Date() }` etc.? Then it would be an imperfect clone.

A JavaScript user must decide on cloning method themselves on a case-by-case basis because it's unknown what they may have there.

Anyway, this caused a bug in my application and it took me a while (4-6h) to debug this because I didn't expect that this function could be possibly trying to clone the matrix values.

**Comment by josdejong** [comment]
Thanks, yes makes sense to just leave the contents untouched. If you want to clone the contents, you can call `math.clone` yourself.

I've made a fix for this in the following PR: #2799. If there are no comments on it I'll merge it within a few days.

**Comment by josdejong** [comment]
Fix is published now in `v11.3.0`
